.github/workflows/ci.yml

```diff
@@ -0,0 +1,139 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+name: CI
```

Comment:
This workflow definitions replaces `pr.yml` and `push.yml` that were mostly the same and included duplicated job definitions.

*Notable Changes*
- no concurrent workflow runs per branch (previous workflows are cancelled automatically)
- previous behavior implemented via `if` conditions
- rephrased `push tests` -> `integration tests` and `pr tests` -> `unit tests`
- added build dependencies to make sure resource expensive integration tests are only run when everything else passes
![image](https://user-images.githubusercontent.com/16404496/150543429-aa29d439-9194-444c-a627-a15ca4657c6a.png)

Let's discuss objections related to these changes in this thread.